src/src/lib.rs

```diff
@@ -763,3 +764,116 @@ fn to_tcp_stream_err(err: std::io::Error) -> tcp_glue::StreamErr {
         ),
     }
 }
+
+#[no_mangle]
+pub extern "C" fn roc_fx_commandStatus(
+    cmd: &command_glue::Command,
+) -> RocResult<(), command_glue::CommandErr> {
+    let args = cmd
+        .args
+        .into_iter()
+        .map(|arg| arg.as_str())
+        .collect::<Vec<_>>();
+    let flat_envs = cmd
+        .envs
+        .into_iter()
+        .map(|env| env.as_str())
+        .collect::<Vec<_>>();
+
+    // Environment vairables must be passed in key=value pairs
+    assert_eq!(flat_envs.len() % 2, 0);
+
+    let mut envs = Vec::with_capacity(flat_envs.capacity() / 2);
+    for chunk in flat_envs.chunks(2) {
+        let key = chunk[0];
+        let value = chunk[1];
+        envs.push((key, value));
+    }
+
+    let result = match cmd.clearEnvs {
+        // Clear environment variables if cmd.clearEnvs set
+        true => std::process::Command::new(cmd.program.as_str())
+            .args(args)
+            .env_clear()
+            .envs(envs)
+            .status(),
+        // Inherit environment variables if cmd.clearEnvs is not set
+        false => std::process::Command::new(cmd.program.as_str())
+            .args(args)
+            .envs(envs)
+            .status(),
+    };
+
+    match result {
+        Ok(status) => {
+            if status.success() {
+                RocResult::ok(())
+            } else {
+                let status_code = status.code().unwrap_or_default();
+                let error = command_glue::CommandErr::ExitStatus(status_code);
+                RocResult::err(error)
+            }
+        }
+        Err(err) => {
+            let str = RocStr::from(err.to_string().borrow());
+            let error = command_glue::CommandErr::IOError(str);
+            RocResult::err(error)
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn roc_fx_commandOutput(
+    cmd: &command_glue::Command,
+) -> RocResult<command_glue::Output, command_glue::CommandErr> {
+    let args = cmd
+        .args
+        .into_iter()
+        .map(|arg| arg.as_str())
+        .collect::<Vec<_>>();
+    let flat_envs = cmd
+        .envs
+        .into_iter()
+        .map(|env| env.as_str())
+        .collect::<Vec<_>>();
+
+    // Environment vairables must be passed in key=value pairs
+    assert_eq!(flat_envs.len() % 2, 0);
+
```

Comment:
Here, you can avoid an allocation by eliding collect and on 841, calling `flat_envs.copied().count()`

Reply:
Done for flat_envs. I'm not sure I follow the comment on 841?